aiida/orm/nodes/node.py

```diff
@@ -235,19 +236,26 @@ def __deepcopy__(self, memo):
         """Deep copying a Node is not supported in general, but only for the Data sub class."""
         raise exceptions.InvalidOperation('deep copying a base Node is not supported')
 
-    def _validate(self) -> bool:
+    def _validate(self) -> None:
         """Validate information stored in Node object.
 
-        For the :py:class:`~aiida.orm.Node` base class, this check is always valid.
+        For the :py:class:`~aiida.orm.Node` base class, this does a basic check on nested keys in case the value of any
+        attribute or extra is a dictionary.
+
         Subclasses can override this method to perform additional checks
         and should usually call ``super()._validate()`` first!
 
         This method is called automatically before storing the node in the DB.
         Therefore, use :py:meth:`~aiida.orm.nodes.attributes.NodeAttributes.get()` and similar methods that
         automatically read either from the DB or from the internal attribute cache.
         """
-        # pylint: disable=no-self-use
-        return True
+        for value in self.base.attributes.all.values():
+            if isinstance(value, dict):
+                validate_attribute_extra_dict_value(value)
+
+        for value in self.base.extras.all.values():
+            if isinstance(value, dict):
```

Comment:
Since this now actually does important checking, it is worse if a subclass overrides it and doesn't invoke the super. I checked the plugins in `aiida-core` and only `TrajectoryData` doesn't call it. Can you add the super call there?

Reply:
Well-spotted! Will do 👍